packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,547 @@
+      final double childHeight = child.computeMinIntrinsicHeight(width);
+      minHeight = max(minHeight, childHeight);
+      child = childParentData.nextSibling;
+    }
+    return minHeight;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
+    double maxHeight = 0.0;
+    final double width = constraints.maxWidth / childCount;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final double childHeight = child.computeMaxIntrinsicHeight(width);
+      maxHeight = max(maxHeight, childHeight);
+      child = childParentData.nextSibling;
+    }
+    return maxHeight;
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return defaultComputeDistanceToHighestActualBaseline(baseline);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _SegmentedControlContainerBoxParentData) {
+      child.parentData = new _SegmentedControlContainerBoxParentData();
+    }
+  }
+
+  void _layoutRects(Function nextChild, RenderBox leftChild, RenderBox rightChild) {
+    RenderBox child = leftChild;
+    double start = 0.0;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final Offset childOffset = new Offset(start, 0.0);
+      childParentData.offset = childOffset;
+      final Rect childRect = new Rect.fromLTWH(start, 0.0, child.size.width, child.size.height);
+      RRect rChildRect;
+      if (child == leftChild) {
+        rChildRect = new RRect.fromRectAndCorners(childRect, topLeft: const Radius.circular(3.0),
+            bottomLeft: const Radius.circular(3.0));
+      } else if (child == rightChild) {
+        rChildRect = new RRect.fromRectAndCorners(childRect, topRight: const Radius.circular(3.0),
+            bottomRight: const Radius.circular(3.0));
+      } else {
+        rChildRect = new RRect.fromRectAndCorners(childRect);
+      }
+      childParentData.surroundingRect = rChildRect;
+      start += child.size.width;
+      child = nextChild();
+    }
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = _kMinSegmentedControlHeight;
+
+    double childWidth;
+    if(constraints.maxWidth.isFinite) {
+      childWidth = constraints.maxWidth / childCount;
+    } else {
+      childWidth = constraints.minWidth / childCount;
+      for(RenderBox child in getChildrenAsList()) {
+        childWidth = max(childWidth, child.getMaxIntrinsicWidth(double.infinity));
+      }
+    }
+
+    RenderBox child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final double boxHeight = child.getMaxIntrinsicHeight(childWidth);
+      maxHeight = max(maxHeight, boxHeight);
+      child = childParentData.nextSibling;
```

Comment:
you can simplify this slightly by removing the childParentData line and just using `child = childAfter(child);`